### PR TITLE
Changed the poms hierarchy in openhab core in order to enable …

### PR DESCRIPTION
**Pull request by VelinYordanov**
... specific configuration for openhab core.

Also added the directory-maven-plugin which will be used to locate static-code-analysis config files in the root folder. See eclipse/smarthome#4240 openhab/static-code-analysis#184

If this gets merged I will also add a config file for the static-code-analysis tool similar to this: openhab/openhab2-addons#2810

Signed-off-by: VelinYordanov <velin.iordanov@gmail.com>

### CODE REVIEW

**Comment by kaikreuzer** [comment]
Thanks.
We might want to combine this restructuring directly with moving the parent poms completely out from this project.
@pfink If I remember correctly, that would be your preferred option: To create a separate repo where we maintain the parent poms that are to be used by the projects - correct?

File: pom.xml [hunk 1]
```diff
@@ -59,21 +66,28 @@
...
-              <dependencies>
-                <dependency>
-                  <groupId>com.itemis.maven.plugins</groupId>
-                  <artifactId>unleash-scm-provider-git</artifactId>
-                  <version>2.1.0</version>
-                </dependency>
-              </dependencies>
+                <groupId>org.commonjava.maven.plugins</groupId>
+                <artifactId>directory-maven-plugin</artifactId>
+                <version>0.2</version>
```
Comment (maggu2810) [inline pom.xml]: Isn't the version already defined in the pluginManagement section of `poms/pom.xml`?

Comment (VelinYordanov) [reply pom.xml]: @maggu2810 Yes, I guess there is no point to repeat the version here. I updated the pull request.

File: pom.xml [hunk 2]
```diff
@@ -59,21 +66,27 @@
...
+                    <dependencies>
+                        <dependency>
+                            <groupId>com.itemis.maven.plugins</groupId>
+                            <artifactId>unleash-scm-provider-git</artifactId>
+                            <version>2.1.0</version>
+                        </dependency>
+                    </dependencies>
+                </plugin>
+            </plugins>
+        </pluginManagement>
```
Comment (pfink) [inline pom.xml]: This block could be removed as the unleash plugin is already available within the parent pom.

**Comment by pfink** [comment]
@kaikreuzer: Yes and no. This would require a little bit more work. My suggestion would be:

- carve out everything that is not belonging to the openHAB versioning scope to a separately versioned parent pom within a seperate repository (e.g. repositories, distribution management, organisational guidelines / plugins like licensing plugin, utility plugins like versioning, ...)
- keep the rest within openhab-core (e.g. openhab version properties)

Basically, the seperate parent pom should always work with **every** openhab version (even older versions). So if I would like to build an older version of openhab I should be able to just update to the newest version of this seperately versioned parent pom and the build should work fine although the infrastructure (like JFrog repositories) may changed meanwhile.

I think for now we could start with this PR if you're OK with it. It would be nice to check it with my sandbox release build if it breaks something up.

**Comment by VelinYordanov** [comment]
@pfink Any further remarks?

**Comment by pfink** [comment]
Not yet. I have to add some options to my release build job and will test if the full release process is not impaired by this. I hope it's OK for you to wait some more days.

**Comment by VelinYordanov** [comment]
@pfink Any progress?

**Comment by pfink** [comment]
@VelinYordanov: The job has no native support for PRs yet, unfortunately :( I "hacked" one build now to go on your branch but it fails with an issue with the Denon binding even though builds on the current openhab master are fine: https://openhab.ci.cloudbees.com/view/Sandbox%20Release%20Pipeline/job/sandbox-openhab2-release/274. Could you provide me another branch where your change is merged with the current openhab-core version?

**Comment by VelinYordanov** [comment]
@pfink Could you provide more detailed explanation of what exactly needs to be done?

**Comment by pfink** [comment]
Ahh, sorry I forgot this one. So you want to merge it now or after the 2.2 release?

**Comment by kaikreuzer** [comment]
Now, if you do not have any objections.

**Comment by pfink** [comment]
As the last release build on this branch was broken, maybe it's not such a good idea. I've overseen the question @VelinYordanov. Tomorrow I'll rebase @VelinYordanov branch on the current openhab-core master and try it again, OK?

**Comment by kaikreuzer** [comment]
Ok, leave your time!

**Comment by pfink** [comment]
Unfortunately, I have to postpone this again because we're currently still facing another issue with the release build...

**Comment by kaikreuzer** [comment]
@pfink Did you find any time for testing that change?

**Comment by kaikreuzer** [comment]
Sorry, closed it by accident...

**Comment by pfink** [comment]
I rebased this on the current openhab-core master and tested it within the release pipeline now, everything was fine: https://openhab.ci.cloudbees.com/view/Sandbox%20Release%20Pipeline/job/sandbox-openhab2-release/319/

PR could be merged from my point of view!

**Comment by kaikreuzer** [comment]
perfect, thanks!

**Pull request merged**